packages/app-mobile/components/ButtonWithTooltip.tsx

```diff
@@ -0,0 +1,197 @@
+//
+// A button with a long-press action. Long-pressing the button displays a tooltip
+//
+
+const React = require('react');
+import { ReactNode } from 'react';
+import { themeStyle } from '@joplin/lib/theme';
+import { Theme } from '@joplin/lib/themes/type';
+import { useState, useMemo, useCallback, useRef } from 'react';
+import {
+	View, Text, Pressable, ViewStyle, PressableStateCallbackType,
+	StyleProp, StyleSheet, LayoutChangeEvent, LayoutRectangle, Animated, AccessibilityState, AccessibilityRole,
+} from 'react-native';
+import { Menu, MenuOptions, MenuTrigger, renderers } from 'react-native-popup-menu';
+
+type ButtonClickListener = ()=> void;
+interface ButtonProps {
+	onClick: ButtonClickListener;
+
+	// Accessibility label and text shown in a tooltip
+	description: string;
+
+	children: ReactNode;
+
+	// [theme] can either be a themeId or a [Theme] object.
+	theme: Theme|number;
+
+
+	style?: ViewStyle;
+	pressedStyle?: ViewStyle;
+	contentStyle?: ViewStyle;
+
+	// Additional accessibility information. See View.accessibilityHint
+	accessibilityHint?: string;
+
+	// Role of the button. Defaults to 'button'.
+	accessibilityRole?: AccessibilityRole;
+	accessibilityState?: AccessibilityState;
+
+	disabled?: boolean;
+}
+
+const ButtonWithTooltip = (props: ButtonProps) => {
+	const [tooltipVisible, setTooltipVisible] = useState(false);
+	const [buttonLayout, setButtonLayout] = useState<LayoutRectangle|null>(null);
+	const tooltipStyles = useTooltipStyles(props.theme);
+
+	// See https://blog.logrocket.com/react-native-touchable-vs-pressable-components/
+	// for more about animating Pressable buttons.
+	const fadeAnim = useRef(new Animated.Value(1)).current;
```

Comment:
I don't think it makes any sense to create a ref, and only keep a reference to .current? It becomes like a regular variable so you could just have `fadeAnim = new Animated...`

Reply:
The React Native documentation was doing it this way (with useRef(...).current). I _think_ this is to prevent callbacks that have the animation as a dependency from being re-created (because different instances of  Animated.Value(1) won't be equal).

Reply:
Ok if that's required to get it working, no problem